src/native/fs/fs_napi.cpp

```diff
@@ -272,7 +273,14 @@ static int
 get_single_user_xattr(int fd, std::string key, std::string& value)
 {
     ssize_t value_len = fgetxattr(fd, key.c_str(), NULL, 0);
-    if (value_len == -1) return -1;
+    if (value_len == -1) {
+        int missing_attr_errno = ENODATA;
+        #ifdef __APPLE__
+            missing_attr_errno = ENOATTR;
+        #endif
+        if (errno != missing_attr_errno) return -1;
+        value = "";
```

Comment:
I'm just not sure if we can have a value of empty string "" which id different from attributes that doesn't exist. Maybe do this error check in get_fd_xattr and then just skip. WDYT?